packages/ember-runtime/lib/system/core_object.js

```diff
@@ -196,11 +196,7 @@ function makeCtor(base) {
             } else if (typeof self.setUnknownProperty === 'function' && !(keyName in self)) {
               self.setUnknownProperty(keyName, value);
             } else {
-              if (DEBUG) {
-                defineProperty(self, keyName, null, value); // setup mandatory setter
-              } else {
-                self[keyName] = value;
-              }
```

Comment:
`defineProperty` defaults to same logic if descriptor argument is `null` as above